tavern/app.go

```diff
@@ -105,45 +106,16 @@ func NewServer(ctx context.Context, options ...func(*Config)) (*Server, error) {
 		createTestData(ctx, client)
 	}
 
-	// Create GraphQL Handler
-	srv := handler.NewDefaultServer(graphql.NewSchema(client))
-	srv.Use(entgql.Transactioner{TxOpener: client})
-
-	// GraphQL Logging
-	gqlLogger := log.New(os.Stderr, "[GraphQL] ", log.Flags())
-	srv.AroundOperations(func(ctx context.Context, next gqlgraphql.OperationHandler) gqlgraphql.ResponseHandler {
-		oc := gqlgraphql.GetOperationContext(ctx)
-		reqVars, err := json.Marshal(oc.Variables)
-		if err != nil {
-			gqlLogger.Printf("[ERROR] failed to marshal variables to JSON: %v", err)
-			return next(ctx)
-		}
-
-		authName := "unknown"
-		id := auth.IdentityFromContext(ctx)
-		if id != nil {
-			authName = id.String()
-		}
-
-		gqlLogger.Printf("%s (%s): %s", oc.OperationName, authName, string(reqVars))
-		return next(ctx)
-	})
-
-	// Setup HTTP Handler
+	// Setup HTTP Handlers
 	router := http.NewServeMux()
 	router.Handle("/status", newStatusHandler())
-	router.Handle("/",
-		playground.Handler("Tavern", "/graphql"),
-	)
-	router.Handle("/graphql", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		w.Header().Set("Access-Control-Allow-Origin", "*")
-		w.Header().Set("Access-Control-Allow-Headers", "*")
-		srv.ServeHTTP(w, req)
-	}))
 	router.Handle("/oauth/login", auth.NewOAuthLoginHandler(cfg.oauth, privKey))
 	router.Handle("/oauth/authorize", auth.NewOAuthAuthorizationHandler(cfg.oauth, pubKey, client, "https://www.googleapis.com/oauth2/v3/userinfo"))
+	router.Handle("/graphql", newGraphQLHandler(client))
```

Comment:
Does this allow un-authed access to `/graphql`?

Reply:
No, we delegate Auth to directives in the GraphQL schema (so some mutations are unauthed)